src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILImporter.Interpreter.cs

```diff
@@ -239,22 +255,1021 @@ private void ImportLoadFloat(double value)
 
         private void ImportShiftOperation(ILOpcode opcode)
         {
-            throw new NotImplementedException();
+            StackItem op1 = PopWithValidation();
+            StackItem op2 = PopWithValidation();
+
+            switch (op1.Kind)
+            {
+                case StackValueKind.Int32:
```

Comment:
There is a lot of duplicated code. 

You should have one `switch` to compute the value of `shiftBy` (that value can be always be int32). Followed by second switch to do the actual shift.

Reply:
This has been done